examples/webgpu_shadow_contact.html

```diff
@@ -0,0 +1,257 @@
+			const PLANE_HEIGHT = 2.5;
+			const CAMERA_HEIGHT = 0.3;
+
+			const state = {
+				shadow: { blur: 3.5, darkness: 1.0, opacity: 1.0 },
+				plane: { color: '#ffffff', opacity: 1.0 },
+				showWireframe: false
+			};
+
+
+			const uBlur = uniform( state.shadow.blur );
+			const uDarkness = uniform( state.shadow.darkness );
+			const uShadowOpacity = uniform( state.shadow.opacity );
+			const uPlaneOpacity = uniform( state.plane.opacity );
+			const uPlaneColor = uniform( new THREE.Color( state.plane.color ) );
+			const uPlaneY = uniform( - 0.3 );
+
+			init();
+
+			function init() {
+
+				camera = new THREE.PerspectiveCamera( 50, window.innerWidth / window.innerHeight, 0.1, 100 );
+				camera.position.set( 0.5, 1, 2 );
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0xffffff );
+
+				window.addEventListener( 'resize', onWindowResize );
+
+				const geometries = [
+					new THREE.BoxGeometry( 0.4, 0.4, 0.4 ),
+					new THREE.IcosahedronGeometry( 0.3 ),
+					new THREE.TorusKnotGeometry( 0.4, 0.05, 256, 24, 1, 3 )
+				];
+
+				const material = new THREE.MeshNormalMaterial();
+
+				for ( let i = 0, l = geometries.length; i < l; i ++ ) {
+
+					const angle = ( i / l ) * Math.PI * 2;
+					const mesh = new THREE.Mesh( geometries[ i ], material );
+					mesh.position.y = 0.1;
+					mesh.position.x = Math.cos( angle ) / 2.0;
+					mesh.position.z = Math.sin( angle ) / 2.0;
+					scene.add( mesh );
+					meshes.push( mesh );
+
+				}
+
+				shadowGroup = new THREE.Group();
+				shadowGroup.position.y = uPlaneY.value;
+				scene.add( shadowGroup );
+
+				renderTarget = new THREE.RenderTarget( 512, 512, { depthBuffer: true } );
+				renderTarget.texture.generateMipmaps = false;
+
+				const planeGeometry = new THREE.PlaneGeometry( PLANE_WIDTH, PLANE_HEIGHT ).rotateX( Math.PI / 2 );
+
+				depthMaterial = new THREE.MeshBasicNodeMaterial();
+
+				const alphaDepth = float( 1 ).sub( depth ).mul( uDarkness );
+
+				depthMaterial.outputNode = vec4( vec3( 0 ), alphaDepth );
+				depthMaterial.depthTest = false;
+				depthMaterial.depthWrite = false;
+
+				shadowPlaneMaterial = new THREE.MeshBasicNodeMaterial();
+				shadowPlaneMaterial.transparent = true;
+				shadowPlaneMaterial.depthWrite = false;
+
+				if ( ! renderTarget.texture.image ) renderTarget.texture.image = { width: 512, height: 512 };
+
+				const blurredShadow = gaussianBlur( texture( renderTarget.texture ), uBlur, 4, { premultipliedAlpha: false } );
+				shadowPlaneMaterial.outputNode = vec4( vec3( 0 ), blurredShadow.a.mul( uShadowOpacity ) );
+
+				plane = new THREE.Mesh( planeGeometry, shadowPlaneMaterial );
+				plane.renderOrder = 1;
+				plane.scale.y = - 1;
+				plane.scale.z = - 1;
+				shadowGroup.add( plane );
+
+				fillPlaneMaterial = new THREE.MeshBasicNodeMaterial();
+				fillPlaneMaterial.transparent = true;
+				fillPlaneMaterial.depthWrite = false;
+				fillPlaneMaterial.outputNode = vec4( uPlaneColor.toVec3(), uPlaneOpacity );
```

Comment:
I think it would be better to use `THREE.NodeMaterial` instead of `MeshBasicNodeMaterial` for these materials; I think we could use `colorNode` instead of `outputNode` as well.

Reply:
Done